src/vllm_tgis_adapter/__main__.py

```diff
@@ -36,15 +36,16 @@ async def start_servers(args: argparse.Namespace) -> None:
     async with build_async_engine_client(args) as engine:
         add_logging_wrappers(engine)
 
+        vllm_server = build_http_server(args)
```

Comment:
There's a small chance for a race condition at startup here, since the fastapi app isn't actually initialized until `await init_app_state(...)` finishes, which is still inside of `run_http_server`.

Probably not the biggest deal, and I think if the readiness checks are using the http server then we won't get traffic until the http server is fully initialized, but it might be best to make `build_http_server` async and move `init_app_state` into it so we can be fully sure the state is initialized before any potential access by the grpc server

Reply:
so, doesn't this race condition already exist because the vllm http server is initialized in an async manner, so it's possible our grpc server comes up before it's ready anyways? if I make `build_http_server` async i dont see how that solves that problem? If we're getting a request to load a lora adapter and the server isn't ready yet, that request is going to fail anyways when it hits the load adapter endpoint on the vllm server. But I guess i can move this over to async and wait until it's init'd

Reply:
Yes, but currently I don't think the grpc server actually depends on the http server in any way. It only depends on the engine to run the requests, and the engine should be ready to go before we build the servers